### PR TITLE
[agent] remove `InstanceParams` class

### PR DESCRIPTION
**Pull request by simonlingoogle**
This commit removes the `InstanceParams` to make the code more clean.

### CODE REVIEW

File: src/backbone_router/nd_proxy.hpp [hunk 1]
```diff
@@ -150,6 +152,7 @@ class NdProxyManager : public MainloopProcessor, private NonCopyable
     int HandleNetfilterQueue(struct nfq_q_handle *aNfQueueHandler, struct nfgenmsg *aNfMsg, struct nfq_data *aNfData);
 
     otbr::Ncp::ControllerOpenThread &mNcp;
+    std::string                      mBackboneInterfaceName;
```
Comment (bukepo) [inline src/backbone_router/nd_proxy.hpp:155]: interface has fixed max length, is it better to use char array?

Comment (simonlingoogle) [reply src/backbone_router/nd_proxy.hpp:155]: The max interface name size is more than 100 bytes, but a typical interface name is only about 10 bytes. It seems a waste of memory to pre-allocate big buffers. 

Moreover, std::string is much more easier to use.

So, I would prefer using std::string as is. 

Thoughts?

Comment (bukepo) [reply src/backbone_router/nd_proxy.hpp:155]: From https://github.com/spotify/linux/blob/master/include/linux/if.h#L26, it's only 16 bytes long.

Comment (simonlingoogle) [reply src/backbone_router/nd_proxy.hpp:155]: > From https://github.com/spotify/linux/blob/master/include/linux/if.h#L26, it's only 16 bytes long.

thanks for the link. 

It's still more convenient to use std::string? 

I don't think there is obvious advantage of using fixed length char arrays.

**Review by bukepo — APPROVED** [review]

File: src/backbone_router/nd_proxy.cpp [hunk 2]
```diff
@@ -377,8 +376,8 @@ otbrError NdProxyManager::InitIcmp6RawSocket(void)
     VerifyOrExit(mIcmp6RawSock >= 0, error = OTBR_ERROR_ERRNO);
 
 #if __linux__
-    VerifyOrExit(setsockopt(mIcmp6RawSock, SOL_SOCKET, SO_BINDTODEVICE, InstanceParams::Get().GetBackboneIfName(),
-                            strlen(InstanceParams::Get().GetBackboneIfName())) == 0,
+    VerifyOrExit(setsockopt(mIcmp6RawSock, SOL_SOCKET, SO_BINDTODEVICE, mBackboneInterfaceName.c_str(),
+                            strlen(mBackboneInterfaceName.c_str())) == 0,
```
Comment (superwhd) [inline src/backbone_router/nd_proxy.cpp:380]: Use `mBackboneInterfaceName.size()`?

**Review by superwhd — APPROVED** [review]
LGTM with a nit 👍

File: src/backbone_router/nd_proxy.cpp (hunk 2 above)
Comment (simonlingoogle) [reply src/backbone_router/nd_proxy.cpp:380]: Good catch

**Review by jwhui — APPROVED** [review]

**Pull request merged**